File: content-solutions/queue-deque/deque-two-door-train.java
import java.util.*;

class Solution {
    public List<Integer> solution(String[] commands) {
        Deque<Integer> train = new ArrayDeque<>();
        for (String command : commands) {
            String[] parts = command.split(" ");
            switch (parts[0]) {
                case "push_front":
                    train.offerFirst(Integer.parseInt(parts[1]));
                    break;
                case "push_back":
                    train.offerLast(Integer.parseInt(parts[1]));
                    break;
                case "pop_front":
                    train.pollFirst();
                    break;
                default:
                    train.pollLast();
            }
        }
        return new ArrayList<>(train);
    }
}

File: content-solutions/queue-deque/deque-two-door-train.py
from collections import deque


def solution(commands):
    train = deque()
    for command in commands:
        parts = command.split()
        name = parts[0]
        if name == "push_front":
            train.appendleft(int(parts[1]))
        elif name == "push_back":
            train.append(int(parts[1]))
        elif name == "pop_front":
            if train:
                train.popleft()
        elif train:
            train.pop()
    return list(train)
